### ISSUE TITLE
Crash in CdiLoggerMultilineEnd and CdiLoggerMultilineGetBuffer unless logging is enabled

### ISSUE THREAD

**Issue by f2bo** [aws/aws-cdi-sdk#66]
There's an issue in the `CdiLoggerMultilineEnd` function which causes it to crash when it unconditionally attempts to free the memory pointed at by  `CdiLogMultilineState->buffer_state_ptr`. This pointer is only initialized when logging has been enabled for the specified component.

https://github.com/aws/aws-cdi-sdk/blob/4a00ff961dea88e98bf7b02c1898c7c3f1dcee34/src/common/src/logger.c#L977-L979

Similarly, `CdiLoggerMultilineGetBuffer` accesses the uninitialized pointer.

https://github.com/aws/aws-cdi-sdk/blob/4a00ff961dea88e98bf7b02c1898c7c3f1dcee34/src/common/src/logger.c#L953-L957

To reproduce:

```C++
#include <cdi_logger_api.h>

int main()
{
    CdiLogMultilineState state;

    CdiLoggerInitialize();
    CdiLogHandle handle = CdiLoggerThreadLogGet();

    // !!!! comment out the following line or change enable to false to crash
    CdiLoggerComponentEnable(handle, CdiLogComponent::kLogComponentEndpointManager, /*enable*/ true);

    CdiLoggerMultilineBegin(handle, CdiLogComponent::kLogComponentEndpointManager, CdiLogLevel::kLogFatal, "SomeFunction", 123, &state);
    CdiLoggerMultiline(&state, "This is a multiline message:");
    CdiLoggerMultiline(&state, "Line ONE.");
    CdiLoggerMultiline(&state, "Line TWO.");
    CdiLoggerMultiline(&state, "Line THREE.");

    // !!!! this will also crash unless the logger component is enabled
    // char* buffer = CdiLoggerMultilineGetBuffer(&state);

    // !!!! this will crash when the logger component is disabled
    CdiLoggerMultilineEnd(&state);
    CdiLoggerShutdown(false);

    return 0;
}
```

And another minor issue related to logging, the documentation for `CdiLoggerThreadLogGet` is incorrect. It seems to have been copied from `CdiLoggerThreadLogSet`.

https://github.com/aws/aws-cdi-sdk/blob/4a00ff961dea88e98bf7b02c1898c7c3f1dcee34/include/cdi_logger_api.h#L214-L220

**Comment by ralfjuen** [comment]
Thanks for reporting this issue. We will include a fix for this issue in our next release.

**Comment by ralfjuen** [comment]
Fixed in v2.3.1.
